app/jobs/v3/service_instance_async_job.rb

```diff
@@ -0,0 +1,174 @@
+require 'jobs/reoccurring_job'
+
+module VCAP::CloudController
+  module V3
+    class ServiceInstanceAsyncJob < VCAP::CloudController::Jobs::ReoccurringJob
+      attr_reader :warnings
+
+      def initialize(guid, audit_info)
+        super()
+        @service_instance_guid = guid
+        @user_audit_info = audit_info
+        @warnings = []
+        @request_attr = {}
+        @first_time = true
+      end
+
+      def perform
+        client = VCAP::Services::ServiceClientProvider.provide({ instance: service_instance })
+
+        check_pre_conditions
+
+        begin
+          if @first_time
+            compute_maximum_duration
+            execute_request(client)
+            compatibility_checks
+            @first_time = false
+          end
+
+          gone! && return if service_instance.blank?
+          operation_in_progress = service_instance.last_operation&.type
+          aborted!(operation_in_progress) if operation_in_progress != operation_type
```

Comment:
Also, for update, this means that we run these checks twice (same code is executed in the check_pre_conditions). The annoying bit is that it yields to different results. As one is captured by the `rescue` block and the other is not. So one will end up changing last operation and the other won't.

Reply:
That's a good point that wasn't being captured in the tests. I'll update that.

And yes for the update behaviour, we'll check it twice, but it will yield the same result: it cannot fail on the second and pass on the first. I'll look into ways to make that a tad better 🤔